### ISSUE TITLE
Lid close does nothing, can only wake up from suspend with power button

### ISSUE THREAD

**Issue by Rev-b** [linux-surface/linux-surface#310]
Hi there.

Giving a new life to my old surface 3 with type cover, using Linux Mint 20 Cinnamon and 5.8.15-surface kernel.

Everything seems to work fine but lid close/open does nothing, no matter what I set in power management

Also, I can only wake up from suspend pressing the power button, that brings me to a locked screen and after typing the password  it goes back to the desktop with "Shut down this system now?" window.

Typecover keyboard or touchpad don't wake it up, nor do external usb mouse or keyboard.

Keyboard lighthing stays on too, it doesn't turn off with suspend.

Am I missing something here?

Thanks.

**Comment by Rev-b** [comment]
Same with 5.9 kernel, updated today.

**Comment by ckj996** [comment]
Same issue with 5.9.6-surface on SL3 Intel.

**Comment by qzed** [comment]
Can you both post the output of `sudo dmidecode -t system`?

**Comment by ckj996** [comment]
```
❯ sudo dmidecode -t system
# dmidecode 3.2
Getting SMBIOS data from sysfs.
SMBIOS 3.2.0 present.

Handle 0x0001, DMI type 1, 27 bytes
System Information
	Manufacturer: Microsoft Corporation
	Product Name: Surface Laptop 3
	Version: 124I:00036T:000M:0300000D:0B:07F:1C:05P:48S:01E:0Y:1K:0U:08
	Serial Number: 040228594057
	UUID: 29a97d2a-bb89-b7f2-9821-c5849dd4dc8d
	Wake-up Type: Unknown
	SKU Number: Surface_Laptop_3_1872
	Family: Surface
```

**Comment by ckj996** [comment]
BTW, mine is business model (15 inch Intel).
link: https://www.microsoft.com/en-us/p/surface-laptop-3-for-business/93n7b0kdzncc?activetab=techspecs

**Comment by qzed** [comment]
@ckj996 Can you
- Clone the repo and checkout the `sl3-1872` branch (`git clone --branch sl3-1872 https://github.com/linux-surface/surface-gpe/`)  and change into the directory (`cd surface-gpe`),
- then build and install the module via `sudo make dkms-install`
- reboot and try suspending and waking up via the lid?

Once you're done you can uninstall the module again by running `sudo make dkms-uninstall` from the repo's directory.

**Comment by qzed** [comment]
> BTW, mine is business model (15 inch Intel).
> link: https://www.microsoft.com/en-us/p/surface-laptop-3-for-business/93n7b0kdzncc?activetab=techspecs

Yeah, gathered that from the SKU. Currently the GPE module only supports the non-business version, but the change I pushed to the repo above should fix that.

**Comment by qzed** [comment]
@Rev-b the above will not work for you. I'll need to do a similar change for the S3 once I've got a dmidecode output for that.

**Comment by ckj996** [comment]
Yes it works after installing the module.

Thank you :)

**Comment by qzed** [comment]
Perfect, you can keep the module installed for now, if you want. I'll include the fix in the next release (could take a bit of time until that is out though).

**Comment by qzed** [comment]
Regarding the Surface 3: The GPE interrupt only notifies the Lid device on close (see [here](https://github.com/linux-surface/acpidumps/blob/master/surface_3/dsdt.dsl#L12088-L12101)) but not on open, so the GPE module won't help here. Closing the lid should work as expected though. How does the device behave on Windows? There is a `Notify` call on the `NTRG` device, so could be that this should trigger the actual wake-up.

**Comment by kitakar5525** [comment]
I haven't tried it myself, but there is a [surface3-wmi driver](https://github.com/torvalds/linux/blob/master/drivers/platform/x86/surface3-wmi.c). According to [the commit message](https://github.com/torvalds/linux/commit/3dda3b3798f96d2974b5f60811142d3e25547807), it requires to configure an udev rule file.

**Comment by Rev-b** [comment]
Hi there
Here it is:

`# dmidecode 3.2
Getting SMBIOS data from sysfs.
SMBIOS 2.8 present.

Handle 0x0001, DMI type 1, 27 bytes
System Information
	Manufacturer: Microsoft Corporation
	Product Name: Surface 3
	Version: B16D1SW1C4G1X1
	Serial Number: 050615451952
	UUID: cbbce68f-e5b5-777b-2c42-5391a6cab7c2
	Wake-up Type: Power Switch
	SKU Number: Surface_3
	Family: Surface

Handle 0x0009, DMI type 12, 5 bytes
System Configuration Options
	Option 1: SMI:00B29C
	Option 2: DSN:                                
	Option 3: DSN:123456789ABCDEF                 
	Option 4: DSN:123456789ABCDEF                 

Handle 0x000B, DMI type 32, 20 bytes
System Boot Information
	Status: No errors detected`

**Comment by qzed** [comment]
@Rev-b Can you try if adding the udev line mentioned in the commit message pointed to by @kitakar5525 above helps?

**Comment by Rev-b** [comment]
> @Rev-b Can you try if adding the udev rule mentioned in the commit message pointed to by @kitakar5525 above helps?

Could you guide me through it?
I know my linux basics, but I'm not sure if I did it right.

I created a file /etc/udev/rules.d/powerswitch.rules and pasted 

SUBSYSTEM=="input", KERNEL=="event*", KERNELS=="surface3-wmi", \
	TAG+="power-switch"

into it.

Rebooted and it did nothing.

**Comment by qzed** [comment]
Hmm, can we rule out any other configuration issues?

**Comment by Rev-b** [comment]
> Hmm, can we rule out any other configuration issues?

Such as?

**Comment by qzed** [comment]
I think Gnome should suspend on lid-close by default, but there are other places where you can configure something like this. For example via `acpid` (if you've installed that) or systemd via `/etc/systemd/logind.conf`. In combination with Gnome, the latter should probably be set to do nothing, but you can probably try changing the systemd config (I'd generally avoid acpid). Having multiple of those configured to do different (or even the same) things might lead to problems.

**Comment by Rev-b** [comment]
logind.cong is

#  This file is part of systemd.
#
#  systemd is free software; you can redistribute it and/or modify it
#  under the terms of the GNU Lesser General Public License as published by
#  the Free Software Foundation; either version 2.1 of the License, or
#  (at your option) any later version.
#
# Entries in this file show the compile time defaults.
# You can change settings by editing this file.
# Defaults can be restored by simply deleting this file.
#
# See logind.conf(5) for details.

[Login]
#NAutoVTs=6
#ReserveVT=6
#KillUserProcesses=no
#KillOnlyUsers=
#KillExcludeUsers=root
#InhibitDelayMaxSec=5
#HandlePowerKey=poweroff
#HandleSuspendKey=suspend
#HandleHibernateKey=hibernate
#HandleLidSwitch=suspend
#HandleLidSwitchExternalPower=suspend
#HandleLidSwitchDocked=ignore
#PowerKeyIgnoreInhibited=no
#SuspendKeyIgnoreInhibited=no
#HibernateKeyIgnoreInhibited=no
#LidSwitchIgnoreInhibited=yes
#HoldoffTimeoutSec=30s
#IdleAction=ignore
#IdleActionSec=30min
#RuntimeDirectorySize=10%
#RemoveIPC=yes
#InhibitorsMax=8192
#SessionsMax=8192

acpid is installed, but I guess this is default in Linux Mint. I didn't mess up with that.

Note that this is a Linux Mint 20 Cinnamon fresh install on a Surface 3 4gb RAM 128 Gb disk, wifi only, using linux-surface kernel. Lid close / open suspend / wake up never worked.

**Comment by qzed** [comment]
Can you try commenting out
```
HandleLidSwitch=suspend
HandleLidSwitchExternalPower=suspend
```
and reboot? If that doesn't do something I guess we can rule config out.

**Comment by Rev-b** [comment]
> Can you try commenting out
> 
> ```
> HandleLidSwitch=suspend
> HandleLidSwitchExternalPower=suspend
> ```
> 
> and reboot? If that doesn't do something I guess we can rule config out.

Did it, lid close / open still does nothing.

**Comment by qzed** [comment]
Hmm okay, no clue then.
